sdk/python/feast/registry.py

```diff
@@ -490,3 +495,72 @@ def _write_registry(self, registry_proto: RegistryProto):
         file_obj.seek(0)
         blob.upload_from_file(file_obj)
         return
+
+
+class S3RegistryStore(RegistryStore):
+    def __init__(self, uri: str):
+        self._uri = urlparse(uri)
+        self._bucket = self._uri.hostname
+        self._key = self._uri.path.lstrip("/")
+        return
```

Comment:
```suggestion
        try:
            import boto3
            import botocore
        except ImportError as e:
            from feast.errors import FeastExtrasDependencyImportError
            raise FeastExtrasDependencyImportError("aws", str(e))
        self.s3 = boto3.resource("s3", endpoint_url=os.environ.get("FEAST_S3_ENDPOINT_URL"))
        return
```
perhaps adapt this pattern https://github.com/feast-dev/feast/blob/73ac3de613a729da41b7c832bb4a520132399339/sdk/python/feast/registry.py#L433-L440